### DISCUSSION TITLE
how to create a middleware to protect the DELETE and POST method for user with the same role?

### DISCUSSION THREAD

**Discussion by Gerguevara** [strapi/strapi#6597] [Legacy Topics]
Hello could you help me please?

I'm trying to create a function to protect a POST and DELETE method.

I have the problem that all authenticated users  have the ability to delete or edit  posts of other users if they know the id of that post.

In my table relationship the user has a one to many relationship with the posts and the posts a relationship that each property has a user

I thought the best way to avoid this is a middleware to extract the Headers token in some way and compare it with the user id that saves each post

but I don't know well the syntax to do it with strapi

I am using the sqlite database

Help me pelase :(

**Comment by derrickmehaffy** [comment]
This would be better handled with a reusable policy that is attached to the route.

https://strapi.io/documentation/v3.x/concepts/policies.html

Basically what you need to check is if the incoming `ctx.state.user.id` matches the same ID on the entry to be modified.
